web/src/user_group_edit.js

```diff
@@ -114,6 +114,31 @@ export function handle_member_edit_event(group_id) {
     const members = [...group.members];
     user_group_edit_members.update_member_list_widget(group_id, members);
 
+    // update display of group-rows on left panel.
+    // We need this update only if your-groups tab is active
+    // and current user is among the affect users as in that
+    // case the group widget list need to be updated and show
+    // or remove the group-row on the left panel accordingly.
+    const tab_key = user_group_settings_ui.get_active_data().$tabs.first().attr("data-tab-key");
+    if (tab_key === "your-groups" && user_ids.includes(people.my_current_user_id())) {
+        user_group_settings_ui.redraw_user_group_list();
+    }
+
+    // update display of check-mark.
+    if (user_group_settings_ui.is_group_already_present(group)) {
+        const is_member = user_groups.is_user_in_group(group_id, people.my_current_user_id());
+        const $sub_unsub_button = user_group_settings_ui
+            .row_for_group_id(group_id)
+            .find(".sub_unsub_button");
+        if (is_member) {
+            $sub_unsub_button.removeClass("disabled");
+            $sub_unsub_button.addClass("checked");
+        } else {
+            $sub_unsub_button.removeClass("checked");
+            $sub_unsub_button.addClass("disabled");
+        }
+    }
```

Comment:
This can probably be moved to `user_group_ui_updates`. Should be fine to do as a follow-up as well. General pattern that I have noticed is - `stream_settings_ui` contains functions which are called on recieving events and then those function calls the appropriate function in `stream_data` to update the data objects and function in `stream_ui_updates` to update the UI.

This pattern is still not followed strictly for stream settings, but we can try doing so here as it sounds like a nice pattern to follow which will help in finding the code easily. We might also make sure we avoid circular imports as we would be converting these modules to `Typescript`.

Reply:
I think user_group modules need significant refactoring as there are lot of cyclic imports and many functions do not align with the idea (that sounds really good) you mentioned, that deserve it's own PR. 
Please comment in case I should complete the refactors and add the commits to this PR only.